### DISCUSSION TITLE
Can sitemaps be generated buildtime with vite?

### DISCUSSION THREAD

**Discussion by phlipsterit** [fedeya/remix-sitemap#66] [General]
This packages looks very nice. But can we use it to generate sitemaps build time when building Remix with Vite? From my understanding this will be a problem because Remix with Vite does not allow custom export from routes which is needed for build time site maps.

When building remix with vite, it gets stricter regarding route exports, [https://remix.run/docs/en/main/future/vite#strict-route-exports](https://remix.run/docs/en/main/future/vite#strict-route-exports)
"So instead of treeshaking, its better to be explicit about what code is client-safe and what code is server-only. For route modules, that means only exporting Remix route exports"

But to generate site maps buildtime for slugs, we need to export a custom function from the route module : 
```
// app/routes/posts.$slug.tsx
import type { SitemapFunction } from 'remix-sitemap';

export const sitemap: SitemapFunction = ({ config, request }) => {
  const posts = await getPosts();
  ...

```

**Comment by voorjordanp** [comment]
Would be interested to know this too

**Comment by sixers** [comment]
@phlipsterit 
Here's I use `remix-sitemap` with Vite:
1. Configure https://github.com/pcattori/vite-env-only plugin
2. Create `posts.sitemap.tsx`file outside of `routes` folder (or use folder with `route.tsx` file and put it next to it), and export the sitemap function like this:
    ```typescript
    import { type SitemapArgs } from "remix-sitemap";
    import { serverOnly$ } from "vite-env-only";
    
    export const sitemap = serverOnly$(async ({ config, request }: SitemapArgs) => {
      // return sitemap
    });
    ```

3. Reexport it in your route:

    ```typescript
    export { sitemap } from "./posts.sitemap";
    ```
